kernel/src/schema.rs

```diff
@@ -122,6 +122,16 @@ impl StructField {
         }
     }
 
+    /// Creates a new nullable field
+    pub fn nullable(name: impl Into<String>, data_type: impl Into<DataType>) -> Self {
+        Self::new(name, data_type, true)
+    }
+
+    /// Creates a new non-nullable field
+    pub fn not_null(name: impl Into<String>, data_type: impl Into<DataType>) -> Self {
+        Self::new(name, data_type, false)
+    }
```

Comment:
maybe (this is a big maybe - might not be useful) we should call out in doc comments either here or in the `new` constructor that these exist just as shortcuts basically?